File: backend/src/main/java/com/onboardingdiary/repository/FeedbackRepository.java
package com.onboardingdiary.repository;

import com.onboardingdiary.entity.Feedback;
import com.onboardingdiary.enums.FeedbackType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.time.LocalDate;
import java.util.UUID;

public interface FeedbackRepository extends JpaRepository<Feedback, UUID> {

    @Query("SELECT f FROM Feedback f WHERE f.user.id = :userId " +
            "AND (:dateFrom IS NULL OR f.date >= :dateFrom) " +
            "AND (:dateTo IS NULL OR f.date <= :dateTo) " +
            "AND (:type IS NULL OR f.type = :type) " +
            "ORDER BY f.date DESC")
    Page<Feedback> findByUserWithFilters(@Param("userId") UUID userId,
                                         @Param("dateFrom") LocalDate dateFrom,
                                         @Param("dateTo") LocalDate dateTo,
                                         @Param("type") FeedbackType type,
                                         Pageable pageable);

    long countByUserId(UUID userId);
}

File: backend/OnboardingDiary.Api/Repositories/FeedbackRepository.cs
using Microsoft.EntityFrameworkCore;
using OnboardingDiary.Api.Data;
using OnboardingDiary.Api.Entities;
using OnboardingDiary.Api.Entities.Enums;

namespace OnboardingDiary.Api.Repositories;

public class FeedbackRepository : Repository<FeedbackEntry>, IFeedbackRepository
{
    private readonly AppDbContext _context;

    public FeedbackRepository(AppDbContext context) : base(context)
    {
        _context = context;
    }

    public IQueryable<FeedbackEntry> GetByUserId(int userId)
    {
        return Query().Where(f => f.UserId == userId);
    }

    public IQueryable<FeedbackEntry> GetByManagerId(int managerId)
    {
        return Query().Where(f => f.User.ManagerId == managerId);
    }

    public IQueryable<FeedbackEntry> GetByDateRange(IQueryable<FeedbackEntry> query, DateTime? dateFrom, DateTime? dateTo)
    {
        if (dateFrom.HasValue)
            query = query.Where(f => f.Date >= dateFrom.Value);

        if (dateTo.HasValue)
            query = query.Where(f => f.Date <= dateTo.Value);

        return query;
    }

    public IQueryable<FeedbackEntry> GetByType(IQueryable<FeedbackEntry> query, FeedbackType? type)
    {
        if (type.HasValue)
            query = query.Where(f => f.Type == type.Value);

        return query;
    }

    public async Task<bool> IsUserManagedByAsync(int entryUserId, int managerId)
    {
        return await _context.Users.AnyAsync(u => u.Id == entryUserId && u.ManagerId == managerId);
    }
}
